.test/test_framework/container/test_utils/aurora_test_utility.go

```
@@ -407,3 +422,167 @@ func RequireTestEnvironmentFeatures(t *testing.T, testEnvironmentRequestFeatures
+		default:
+			return nil, fmt.Errorf("unsupported database engine: %v", engine)
+		}
+	case RDS_MULTI_AZ_CLUSTER:
+		switch engine {
+		case MYSQL:
+			// For Multi-AZ MySQL, we need to get the replica writer ID first
+			retrieveTopologySQL = `SELECT SUBSTRING_INDEX(endpoint, '.', 1) as SERVER_ID
+								   FROM mysql.rds_topology
+								   ORDER BY CASE WHEN id = @@server_id THEN 0 ELSE 1 END,
+								            SUBSTRING_INDEX(endpoint, '.', 1)`
+		case PG:
+			retrieveTopologySQL = `SELECT SUBSTRING(endpoint FROM 0 FOR POSITION('.' IN endpoint)) as SERVER_ID 
+								   FROM rds_tools.show_topology() 
+								   ORDER BY CASE WHEN id = (SELECT MAX(multi_az_db_cluster_source_dbi_resource_id) 
+								                            FROM rds_tools.multi_az_db_cluster_source_dbi_resource_id()) 
+								                 THEN 0 ELSE 1 END, endpoint`
+		default:
+			return nil, fmt.Errorf("unsupported database engine: %v", engine)
+		}
+	case RDS_MULTI_AZ_INSTANCE:
+		switch engine {
+		case MYSQL:
+			retrieveTopologySQL = "SELECT SUBSTRING_INDEX(endpoint, '.', 1) as SERVER_ID FROM mysql.rds_topology"
+		case PG:
+			retrieveTopologySQL = "SELECT SUBSTRING(endpoint FROM 0 FOR POSITION('.' IN endpoint)) as SERVER_ID FROM rds_tools.show_topology()"
+		default:
+			return nil, fmt.Errorf("unsupported database engine: %v", engine)
+		}
+	default:
+		return nil, fmt.Errorf("unsupported database engine deployment: %v", deployment)
+	}
+
+	// Create connection to execute the query
+	dsn := fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", username, password, clusterEndpoint, port, dbName)
+	if engine == PG {
+		dsn = fmt.Sprintf("postgres://%s:%s@%s:%d/%s?sslmode=require", username, password, clusterEndpoint, port, dbName)
+	}
+
+	db, err := OpenDb(engine, dsn)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to database: %w", err)
+	}
+	defer func(db *sql.DB) {
+		_ = db.Close()
+	}(db)
```

curious, is there a difference between this and defer db.Close()?

---

This is a way to explicitly ignore the error - instead of db.Close()